### PR TITLE
ON-40098 # Changed drafts to use versioning

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: src/draft-service.ts [hunk 1]
```diff
@@ -416,38 +477,65 @@ export async function deleteDraft(
...
+          draftSubmission.formSubmissionDraftId === formSubmissionDraftId,
+      )
+      if (!draftSubmission) {
+        return
+      }
+
+      localDraftsStorage.unsyncedDraftSubmissions =
+        localDraftsStorage.unsyncedDraftSubmissions.filter(
+          (draftSubmission) =>
+            draftSubmission.formSubmissionDraftId !== formSubmissionDraftId,
```
Comment (divporter) [inline src/draft-service.ts:515]: You could just filter it anyway, instead of finding and filtering.

Comment (divporter) [inline src/draft-service.ts:515]:
```suggestion
      localDraftsStorage.unsyncedDraftSubmissions =
        localDraftsStorage.unsyncedDraftSubmissions.filter(
          (draftSubmission) =>
            draftSubmission.formSubmissionDraftId !== formSubmissionDraftId,
```

File: src/draft-service.ts [hunk 2]
```diff
@@ -416,38 +477,65 @@ export async function deleteDraft(
...
+        formSubmissionDraftId,
+        localDraftsStorage,
+      })
+      const draftSubmission = localDraftsStorage.unsyncedDraftSubmissions.find(
+        (draftSubmission) =>
+          draftSubmission.formSubmissionDraftId === formSubmissionDraftId,
+      )
+      if (!draftSubmission) {
+        return
+      }
```
Comment (divporter) [inline src/draft-service.ts:510]:
```suggestion
```

Comment (mymattcarroll) [reply src/draft-service.ts:510]: Notice the `return`, want to stop and not finish function if draft does not exist.

File: src/draft-service.ts [hunk 3]
```diff
@@ -362,31 +430,23 @@ export async function getDrafts(): Promise<SubmissionTypes.FormsAppDraft[]> {
...
   }
 
-  const draftSubmission = await getDraftData(draft.formId, draft.draftDataId)
-
-  return {
-    draftData: draftSubmission.submission,
-    lastElementUpdated: draftSubmission.lastElementUpdated,
-    draft,
-  }
+  return await getDraftSubmission(formSubmissionDraft)
```
Comment (divporter) [inline src/draft-service.ts:449]: I know it's current behaviour, but if you've had the app open for ages you may not necessarily have the latest server data.  Unless I'm mistaken.

File: src/draft-service.ts (hunk 2 above)
Comment (divporter) [reply src/draft-service.ts:510]: I don't see any negative side effects from continuing

File: src/draft-service.ts (hunk 3 above)
Comment (mymattcarroll) [reply src/draft-service.ts:449]: yep yep, will need to be fixed.

**Pull request merged**